src/components/Controls.vue

```diff
@@ -104,6 +216,12 @@ export default Vue.extend({
 
       legendSVG.select('.legendLinear').call(legendLinear);
     },
+    clickButton(this: any) {
+      // call the supergraph function to generate the new nodes and the new edges for the super graph
+      // assign to a variable that can update the network which updates the visualization
+      const superNetwork = superGraph(this.network.nodes, this.network.links);
+      this.network = superNetwork;
```

Comment:
A general note about comments: you shouldn't self-describe the "obvious" parts of what the code is doing. In this case, you don't need to say you're calling a function and assigning to a variable. I'd simplify things like this:

```suggestion
      // Compute a new graph based on aggregating California airports into a supernode.
      this.network = superGraph(this.network.nodes, this.network.links);
```